controller/setting_controller.go

```diff
@@ -223,10 +223,44 @@ func (sc *SettingController) syncSetting(key string) (err error) {
 		if err := sc.syncUpgradeChecker(); err != nil {
 			return err
 		}
+		return nil
 	case string(types.SettingNameBackupTarget), string(types.SettingNameBackupTargetCredentialSecret), string(types.SettingNameBackupstorePollInterval):
 		if err := sc.syncBackupTarget(); err != nil {
 			return err
 		}
+		return nil
+	case string(types.SettingNameKubernetesClusterAutoscalerEnabled):
+		if err := sc.updateKubernetesClusterAutoscalerEnabled(); err != nil {
+			return err
+		}
+		return nil
+	case string(types.SettingNameSupportBundleFailedHistoryLimit):
+		if err := sc.cleanupFailedSupportBundles(); err != nil {
+			return err
+		}
+		return nil
+	case string(types.SettingNameLogLevel):
+		if err := sc.updateLogLevel(); err != nil {
+			return err
+		}
+		return nil
+	case string(types.SettingNameV2DataEngine):
+		if err := sc.updateV2DataEngine(); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	// SettingCategoryDangerZone and they will try to delete pods to update the settings.
+	volumesDetached, err := sc.ds.AreAllVolumesDetached()
```

Comment:
Please have a new `syncSettingForSystemManagedComponents`. 

Besides remaining settings here, also move `sc.updateV2DataEngine` there, but probably you need to refactor it a bit.

P.S. need to wait for https://github.com/longhorn/longhorn-manager/pull/2386 to merge first.

Reply:
Modified to have a new function `syncSettingForSystemManagedComponents`